### DISCUSSION TITLE
!info.txt files showing game items as blank

### DISCUSSION THREAD

**Discussion by mharmless** [Kalanyr/gogrepo#94] [Q&A]
Multiple games are just showing as no game items. Stellaris hasn't updated since 3.7.4, now shows blank items, and new purchases like against the storm download nothing at all.

python gogrepoc.py update -ids against_the_storm

fetches game details just fine, no errors reported. but downloading will show nothing to download presumably because !info.txt says so.

1. 
2. -- Against the Storm --
3. 
4. title.......... against_the_storm
5. genre.......... Strategy
6. game id........ 1460447406
7. url............ https://www.gog.com/en/game/against_the_storm
8. release date... November 01, 2022
9. 
10. game items.....:
11. 
12. 
13. changelog......:
14. 
15. #### Hotfix 0.43.3 (February 3rd, 2023)

changlog cut short here. And this seems to be the problem for a laundry list of games

1. 
2. -- Age of Wonders 4 --
3. 
4. title.......... age_of_wonders_4
5. genre.......... Strategy
6. game id........ 1915432764
7. url............ https://www.gog.com/en/game/age_of_wonders_4
8. release date... May 02, 2023
9. 
10. game items.....:

1. 
2. -- Stellaris --
3. 
4. title.......... stellaris
5. genre.......... Strategy
6. game id........ 1508702879
7. url............ https://www.gog.com/en/game/stellaris
8. user rating.... 78%
9. release date... October 01, 2018
10. 
11. game items.....:
12. 
13. 
14. changelog......:

**Comment by Kalanyr** [comment]
info.txt doesn't control anything, it's just a report that's generated
before downloads are done so that you know what's supposed to be in the
folder if you download it.

This sounds like something is going wrong in the update step. Potentially
not properly detecting the OS / Language versions it should be downloading.

Which version of Gogrepoc are you using ? Master or Dev ?

What OS are you running gogrepo on ?  Are you running it on a device that
might be behaving weirdly (eg some NAS's don't actually report a locale ) ?

And in case I need to walk you through some troubleshooting how familiar
would you say you are with Python ?

**Reply by mharmless** [reply]
The python script shows v0.4.0-a, and I believe it was downloaded from the master updated 8 months ago.
I'm running it on a custom storage server that hosts media and game files. Off the shelf hardware, a basic install of linux mint 21, along with snapraid and mergerfs to handle the storage.

I have done a small amount of scripting in a professional environment, but virtually nothing in python specifically. I can probably follow troubleshooting directions.

More info on the way this runs; a weekly cron job runs a script to update games, and after noticing that stellaris hadn't updated and some new purchases were missing I've been poking at it with a remote desktop session. I'm not sure how long it has been failing to download content.

This is the script that runs, very straightforward.
```
python gogrepoc.py update
python gogrepoc.py download /mnt/fakeraid/Games/__GoG\ Archives/ -os windows -lang en -skipfiles patch_* *_demo_* *.dmg *32bit*
```

Manual running update initially showed 12 updates found, same as the gog website showed. Nothing was downloaded anyway with a subsequent download run, although update no longer claims to see new content so it changed something.

If I run without the os restriction, it does try to download stellaris 3.10.3 linux version, so it sees that one at least. If I run without skipfiles or lang, no change.

If I remove a previous version, such as 3.7.4 which was most recently downloaded, it does not attempt to redownload that; it will still say nothing to download.

**Comment by Kalanyr** [comment]
The problem appears to be that you're letting update autodetermine what os
and language you're using ( which will probably be Linux for the OS ) but
specifying to download only windows.

You should run the update command with the same -os / -lang you use for
download.

However given how out of date the windows info in your manifest is you
probably need to disable the chron, manually run the following command:


python gogrepoc.py update -os windows -lang en -full

It should take about an hour for every 300 games you own.

Then you need to manually run your download command, that might take
upwards of a day to finish depending on how many missing windows
updates you have ( which is why we needed to disable the chron jobs
for this )

Once that's done, I recommend you make use of the verify command:

python gogrepoc.py verify /mnt/fakeraid/Games/__GoG\ Archives/ -clean
-os windows

Which should use the md5 data from GOG to make sure your files are
correct, it should move any wrong ones to a subdirectory called
!orphaned

That could take a long time if you have a lot of game files.

Let me know what verify outputs when it's down and I'll let you know
how to proceed from there ( I think there's a reasonable chance
there'll be a few issues given how out of date things probably are )

( Verify is safe to use in a chron usually because it remembers files
that it's already verified so subsequent runs are much faster )


On Tue, 12 Dec 2023, 05:56 mharmless, ***@***.***> wrote:

The python script shows v0.4.0-a, and I believe it was downloaded from the

**Comment by Kalanyr** [comment]
For the verify outputs we can just start with the little summary at the
end, you don't have to send me the full thing.

**Reply by mharmless** [reply]
Doing an update with -os windows and -ids stellaris followed by a download has it downloading the 3.10.3 version right now, so this looks to be exactly the problem. I must have run this in the past at some point specifying to download all OS data. For some reason I just assumed update grabbed all possible files and it was up to download to filter what was wanted from the pile. Thank you for helping me see the error.

Manually tested against_the_storm, one of the new purchases, and same deal as stellaris it appears to work just fine now that update has the correct options to pull the right data.

Full update is running now, will check back in on it later today and start the downloads. There are ~700 games in the library so it will take a bit to bring back the verify data, and in fact may end up having to move into January before finishing due to monthly data limits depending on how that looks.

**Comment by Kalanyr** [comment]
Awesome, glad I could help. Keep me posted on the progress.  If you need to
leave more than a couple of days between the downloads you should run a
standard update before doing more downloading too ( otherwise you're pretty
much guaranteed to be trying to download files that have since been removed
and replaced with new versions ).

**Reply by mharmless** [reply]
740 gigs to download, will definitely have to split that across the monthly limit reset.

**Comment by mharmless** [comment]
22:11:36 | --totals------------
22:11:36 | known items......... 1853
22:11:36 | pre-verified items.. 2
22:11:36 | have items.......... 1811
22:11:36 | skipped items....... 454
22:11:36 | missing items....... 42
22:11:36 | md5 mismatches...... 0
22:11:36 | size mismatches..... 0
22:11:36 | zipfile failures.... 0
22:11:36 | cleaned items....... 0
22:11:36 | --
22:11:36 | total time: 4:49:24.787877
22:11:36 | exiting...

Summary looks good, the missing items are 32bit installers, I had a typo in my skipfiles entry when I typed in the verify command. Shocked at how relatively fast the verification was considering just how many games are in the set.

Thank you again for your help, and have a good holiday season.

**Comment by Kalanyr** [comment]
That's excellent, as I side not I recommend changing your chron to:

update
download
clean
verify

or

update
clean
download
verify

( there's trade offs either way, cleaning second is safer but can you use
extra bandwidth when there's only small changes to a file that can be
healed with the md5 chunk data , whereas cleaning second prioritises saving
bandwidth but can require an additional run of the download if the existing
files end up failing it's checks )
